### PR TITLE
Add CI workflow to do local scratch deployment

### PR DESCRIPTION
**Pull request by arwer13**
The general purpose is to keep the scratch deployment scripts alive and sound - this PR is the first iteration.

Namely it add github workflow which spawns local clean anvil node and runs local scratch deploy script and scratch-acceptance-test script on it.

Also:
- ⚠️ add MANAGE_NODE_OPERATOR_ROLE to Agent (as it is on mainnet) upon deployment to simplify and don't add voting-related code to  scratch-acceptance-test. 
- bump github actions:
  - checkout v3 -> v4
  - setup-node v3 -> v4
- bump NodeJS version 16 -> 18
- fix running the simple acceptance test when forking a local debug node (anvil/hardhat/...)
- fix storage-layout action check
  - PR to storage-layout-action https://github.com/lidofinance/storage-layout-action/pull/4
- fix assert-bytecode action check. **NB**: Bytecode checks are disabled in the last commit (as they were, because they are supposed to be enabled only for PRs to master ). To see the result see CI checks at the previous commit [a0dcbdf](https://github.com/lidofinance/lido-dao/pull/807/commits/a0dcbdfc94f3d70ccc5bf7f7e7d7d264a0c7342a)
- remove obsolete github flow
- remove artifacts.json and do generate it only on CI in the corresponding action

### CODE REVIEW

**Comment by TheDZhon** [comment]
Huge :eyeglasses:

**Review by TheDZhon — APPROVED** [review]
LGTM, a single comment only is to consider :pray:

File: contracts/0.4.24/template/LidoTemplate.sol [hunk 1]
```diff
@@ -608,7 +608,12 @@ contract LidoTemplate is IsContract {
         // NodeOperatorsRegistry
         perms[0] = _state.operators.MANAGE_SIGNING_KEYS();
         perms[1] = _state.operators.SET_NODE_OPERATOR_LIMIT_ROLE();
-        for (i = 0; i < 2; ++i) {
+        // Need MANAGE_NODE_OPERATOR_ROLE to make current scratch acceptance script
+        // (scripts/scratch/checks/scratch-acceptance-test.js) work.
+        // After voting utilities added to the repo should rewrite the acceptance
+        // script to grant/renounce this role and remove granting from here.
```
[the remark this answers is not in the document]
Comment (arwer13) [reply contracts/0.4.24/template/LidoTemplate.sol]: done

**Review by TheDZhon — APPROVED** [review]
LGTM

**Pull request merged**